### PR TITLE
[DateInput] Don't closeOnSelection when the time changes 

### PR DESCRIPTION
**Pull request by cmslewis**
#### Checklist

- [x] Include tests

#### Changes proposed in this pull request:

**Before:** incrementing or otherwise changing the time would immediately close the popover if `closeOnSelection=true`.
**After:** Stop doing that. Popover now closes only after clicking a date in the calendar, not after changing the time (or the month, but that was already implemented).

### CODE REVIEW

**Comment by cmslewis** [comment]
@giladgray this is ready for a CR

File: packages/datetime/src/dateInput.tsx [hunk 1]
```diff
@@ -275,10 +275,14 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
     }
 
     private handleDateChange = (date: Date, hasUserManuallySelectedDate: boolean) => {
+        const prevMomentDate = this.state.value;
         const momentDate = fromDateToMoment(date);
-        const hasMonthChanged = date !== null && !isMomentNull(this.state.value) && this.state.value.isValid() &&
-            momentDate.month() !== this.state.value.month();
-        const isOpen = !(this.props.closeOnSelection && hasUserManuallySelectedDate && !hasMonthChanged);
+
+        const isOpen = !(this.props.closeOnSelection
```
Comment (giladgray) [inline packages/datetime/src/dateInput.tsx]: ugh boolean logic. this not-wrapper is throwing my brain for a loop, very hard to follow what's going on here. is there a simpler equivalent expression? #truthtables

**Comment by cmslewis** [comment]
@giladgray can you CR this again?

File: packages/datetime/src/dateInput.tsx [hunk 2]
```diff
@@ -287,6 +291,23 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
         Utils.safeInvoke(this.props.onChange, date === null ? null : fromMomentToDate(momentDate));
     }
 
+    private shouldCheckForDateChanges(prevMomentDate: moment.Moment, nextMomentDate: moment.Moment) {
+        return nextMomentDate !== null && !isMomentNull(prevMomentDate) && prevMomentDate.isValid();
```
Comment (cmslewis) [inline packages/datetime/src/dateInput.tsx]: Copied this from the code above, but definitely seems worth changing. Will do. (https://github.com/palantir/blueprint/pull/1092/files#diff-2899fd9fc14d599c6e4062f791d99120L279)
